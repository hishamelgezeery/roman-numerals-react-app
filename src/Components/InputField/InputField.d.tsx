import { Component } from 'react'

export interface InputFieldProps {
 /**
 * The label text of the input field
 */
 labelText: String;
 /**
 * Boolean to render button with the input field
 */
 withButton: Boolean;
 /**
 * The text of the button
 */
 buttonText: String;
 /**
 * The event handler for button onClick event
 */
 onButtonClick: Function;
 /**
 * The event handler for input field text change event
 */ 
 onInputChange: Function;
 /**
 * The text value of the input field
 */ 
 value: String;
}

export default class InputField extends Component<InputFieldProps> { }
