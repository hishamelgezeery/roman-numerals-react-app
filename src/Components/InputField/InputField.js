import React from "react";
import { Button, InputGroup, FormControl } from "react-bootstrap";

const InputField = ({
  labelText,
  withButton,
  buttonText,
  onButtonClick,
  onInputChange,
  value,
}) => {
  return (
    <>
      <div className="col-sm-3">
        <label>{labelText}</label>
      </div>
      <div className="col-sm-9">
        <InputGroup>
          <FormControl onInput={onInputChange} value={value} />
          {withButton && (
            <InputGroup.Append>
              <Button variant="outline-secondary" onClick={onButtonClick}>
                {buttonText}
              </Button>
            </InputGroup.Append>
          )}
        </InputGroup>
      </div>
    </>
  );
};

export default InputField;
