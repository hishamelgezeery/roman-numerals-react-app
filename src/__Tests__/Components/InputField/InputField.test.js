import renderer from 'react-test-renderer'
import InputField from '../../../Components/InputField'

describe('InputField Component', () => {
  it('should render correctly', () => {
    const instance = renderer.create(<InputField labelText="Test Label"
    value={'test string in input'}
        />).toJSON()
    expect(instance).toMatchSnapshot()
    })
  it('should render correctly with button', () => {
    const instance = renderer.create(<InputField labelText="Test Label"
    withButton
    buttonText="Convert"
    value={'test string in input'}
        />).toJSON()
    expect(instance).toMatchSnapshot()
  })
})
