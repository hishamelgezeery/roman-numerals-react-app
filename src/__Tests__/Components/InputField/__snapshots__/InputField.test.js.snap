// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Component should render correctly 1`] = `
Array [
  <div
    className="col-sm-3"
  >
    <label>
      Test Label
    </label>
  </div>,
  <div
    className="col-sm-9"
  >
    <div
      className="input-group"
    >
      <input
        className="form-control"
        value="test string in input"
      />
    </div>
  </div>,
]
`;

exports[`InputField Component should render correctly with button 1`] = `
Array [
  <div
    className="col-sm-3"
  >
    <label>
      Test Label
    </label>
  </div>,
  <div
    className="col-sm-9"
  >
    <div
      className="input-group"
    >
      <input
        className="form-control"
        value="test string in input"
      />
      <div
        className="input-group-append"
      >
        <button
          className="btn btn-outline-secondary"
          disabled={false}
          type="button"
        >
          Convert
        </button>
      </div>
    </div>
  </div>,
]
`;
